Adds recipe about using msw to mock API endpoints (#370)
miyagi-dev/miyagi: docs/recipes/mocking-api-endpoints.md

# Mocking API endpoints

> Since _miyagi_ is a lot about developing components independently of a backend framework, we might also want to mock API endpoints, for example to test form submissions using AJAX requests.

Reviewer: ```suggestion
Since _miyagi_ is focused on developing components independently of a backend framework, we might also want to mock API endpoints, for example, to test form submissions using AJAX requests.
```

> We can do that easily using [MSW](https://mswjs.io).

Reviewer: ```suggestion
We can do that easily using [Mock Service Worker (MSW)](https://mswjs.io).
```

> Please follow the instructions on the _MSW_ website to setup MSW using _Browser Integration_ (not the Node integration). When generating the worker script, we can use whatever directory you like for the `PUBLIC_DIR`. We will for now use the current directory, so `.`. When we are done with the setup, we will have with the following files:

Reviewer: How about linking to the appropriate place in the docs?

```suggestion
Please follow the [instructions on the _MSW_ website](https://mswjs.io/docs/integrations/browser) to setup MSW using _Browser Integration_ (not the Node integration). When generating the worker script, use whichever directory you prefer for the `PUBLIC_DIR`. We will use the current directory, so `.` for this example. When we are done with the setup, we will have the following files:
```
